test/unit/eigensolver/mc/test_reduction_to_band.cpp

```diff
@@ -0,0 +1,320 @@
+  for (SizeType k = 0; k < matrix.nrTiles().rows(); ++k) {
+    const GlobalTileIndex kk{k, k};
+    const auto& tile = matrix.tile(kk);
+
+    mirror_on_diag(tile);
+  }
+}
+
+template <class T>
+auto all_gather_taus(const SizeType k, const SizeType chunk_size, const SizeType band_size,
+                     std::vector<hpx::shared_future<common::internal::vector<T>>> fut_local_taus,
+                     comm::CommunicatorGrid comm_grid) {
+  using namespace dlaf::comm::sync;
+
+  std::vector<T> taus;
+  taus.reserve(to_sizet(k));
+
+  hpx::wait_all(fut_local_taus);
+  auto local_taus = hpx::unwrap(fut_local_taus);
+
+  DLAF_ASSERT(band_size == chunk_size, band_size, chunk_size);
+
+  const auto n_chunks = std::ceil(float(k) / chunk_size);
+
+  for (auto index_chunk = 0; index_chunk < n_chunks; ++index_chunk) {
+    const auto owner = index_chunk % comm_grid.size().cols();
+    const bool is_owner = owner == comm_grid.rank().col();
+
+    const auto this_chunk_size = std::min(k - index_chunk * chunk_size, chunk_size);
+
+    std::vector<T> chunk_data;
+    if (is_owner) {
+      const auto index_chunk_local = to_sizet(index_chunk / comm_grid.size().cols());
+      chunk_data = local_taus[index_chunk_local];
+      broadcast::send(comm_grid.rowCommunicator(),
+                      common::make_data(chunk_data.data(), static_cast<SizeType>(chunk_data.size())));
+    }
+    else {
+      chunk_data.resize(to_sizet(this_chunk_size));
+      broadcast::receive_from(owner, comm_grid.rowCommunicator(),
+                              common::make_data(chunk_data.data(),
+                                                static_cast<SizeType>(chunk_data.size())));
+    }
+
+    // copy each chunk contiguously
+    std::copy(chunk_data.begin(), chunk_data.end(), std::back_inserter(taus));
+  }
+
+  return taus;
+}
+
+TYPED_TEST(ReductionToBandTest, Correctness) {
+  constexpr Device device = Device::CPU;
+
+  for (auto&& comm_grid : this->commGrids()) {
+    for (const auto& size : square_sizes) {
+      for (const auto& block_size : square_block_sizes) {
+        const SizeType band_size = block_size.rows();
+        const SizeType band_size_tiles = band_size / block_size.rows();
+        const SizeType k_reflectors = size.rows() - band_size;
+
+        Distribution distribution({size.rows(), size.cols()}, block_size, comm_grid.size(),
+                                  comm_grid.rank(), {0, 0});
+
+        // setup the reference input matrix
+        Matrix<const TypeParam, device> reference = [&]() {
+          Matrix<TypeParam, device> reference(distribution);
+          matrix::util::set_random_hermitian(reference);
+          return reference;
+        }();
+
+        Matrix<TypeParam, device> matrix_a(std::move(distribution));
+        copy(reference, matrix_a);
+
+        // Apply reduction-to-band
+        DLAF_ASSERT(band_size == matrix_a.blockSize().rows(), "not yet implemented");
```

Comment:
what about adding some more info? like `band_size`, `matrix_a.blockSize().rows()` o specifying "band sizes smaller than blockSize rows are not yet supported"?

Reply:
I've opted for changing it according to this comment. Is it ok for you?
https://github.com/eth-cscs/DLA-Future/pull/270#discussion_r673961047